extensions/interactions/rules.json

```diff
@@ -0,0 +1,186 @@
+{
+  "CodeRepl": {
```

Comment:
One sec -- are you talking here about putting the rule descriptions etc. in the interaction Python files, or creating a separate rules.json for each interaction? The former won't work: I have a test in the frontend (see rulesSpec.js) that checks that each rules service in the frontend corresponds exactly to the rules defined in the JSON file. Testing this requires that the frontend be able to access the rule definitions without the App Engine server running, so putting the definitions in the Python file will not work.

Also, @wxyxinyu and I discussed this and we realized that we can have a separate rules.json file for each interaction after all. The only thing that's needed is a canonical list of interactions whose validity we should be checking, so we plan to pull out feconf.ALLOWED_INTERACTION_CATEGORIES into a JSON file as well, and use that in the Karma test to construct the URLs for the individual rules.json files.

However, I think I should check that this sounds good before going ahead and doing it. What are your thoughts?